test/python/test_eval_json.py

```diff
@@ -0,0 +1,232 @@
+"""Test the evaluation of JSON results."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import pytest
+
+from mqt.core.evaluation import __aggregate, __flatten_dict, __post_processing
+
+if TYPE_CHECKING:
+    from pytest_console_scripts import ScriptRunner
+
+path_base = Path(__file__).resolve().parent / "results_baseline.json"
+path_feature = Path(__file__).resolve().parent / "results_feature.json"
+
+
+def test_flatten_dict() -> None:
+    """Test flatten_dict."""
+    d1 = {"a": {"b": {"main": 1}}}
+    assert __flatten_dict(d1) == {"a.b.main": 1}
+    d2 = {"a": {"b": {"main": 1, "447-add-benchmark-suite-in-mqt-core": 2}}, "d": {"main": 2}}
+    assert __flatten_dict(d2) == {"a.b.main": 1, "a.b.447-add-benchmark-suite-in-mqt-core": 2, "d.main": 2}
```

Comment:
Also triggers the inner monk in me: can we please get rid of the cryptic branch name here?
```suggestion
    d2 = {"a": {"b": {"main": 1, "feature": 2}}, "d": {"main": 2}}
    assert __flatten_dict(d2) == {"a.b.main": 1, "a.b.feature": 2, "d.main": 2}
```